rc/filetype/markdown.kak

```diff
@@ -41,19 +50,25 @@ add-highlighter shared/markdown regions
 add-highlighter shared/markdown/inline default-region regions
 add-highlighter shared/markdown/inline/text default-region group
 
-evaluate-commands %sh{
-  languages="
-    awk c cabal clojure coffee cpp crystal css cucumber d diff dockerfile elixir erlang fish
-    gas go haml haskell html ini java javascript json julia kak kickstart
-    latex lisp lua makefile markdown moon objc ocaml perl pug python ragel
-    ruby rust sass scala scss sh swift toml tupfile typescript yaml sql
-  "
-  for lang in ${languages}; do
-    printf 'add-highlighter shared/markdown/%s region -match-capture ^(\h*)```\h*(%s\\b|\\{[.=]?%s\\})   ^(\h*)``` regions\n' "${lang}" "${lang}" "${lang}"
-    printf 'add-highlighter shared/markdown/%s/ default-region fill meta\n' "${lang}"
-    [ "${lang}" = kak ] && ref=kakrc || ref="${lang}"
-    printf 'add-highlighter shared/markdown/%s/inner region \A```[^\\n]*\K (?=```) ref %s\n' "${lang}" "${ref}"
-  done
+# Used for the pluggable infrastructure of languages
+define-command -hidden markdown-build-fenced-highlighters %{
+    add-highlighter -override shared/markdown/fenced region -match-capture ^(\h*)``` ^(\h*)``` regions
+    add-highlighter shared/markdown/fenced/default default-region fill meta
+    evaluate-commands %sh{
+      for kv in ${kak_opt_markdown_supported_languages}; do
+        ref=${kv##*=}
+        lang=${kv%%=*}
+        [ -n "$ref" ] || ref=$lang
+        printf 'add-highlighter shared/markdown/fenced/%s region -match-capture ^(\h*)```\h*(%s\\b|\\{[.=]?%s\\})   ^(\h*)``` regions\n' "${lang}" "${lang}" "${lang}"
```

Comment:
I think we need `add-highlighter -override` or else it will error if we modify the option.